aiidalab/app.py

```diff
@@ -131,6 +136,7 @@ def from_id(
         app_id: str,
         registry_entry: dict[str, Any] | None = None,
         apps_path: str | None = None,
+        registered: bool | None = None,
```

Comment:
We should probably use the functools.cached_property decorator.

https://docs.python.org/3/library/functools.html#functools.cached_property